pkg/cmd/infra/router/router.go

```diff
@@ -26,28 +27,104 @@ This command launches a router connected to your OpenShift master. The router li
 created by users and keeps a local router configuration up to date with those changes.`
 )
 
-type templateRouterConfig struct {
-	Config             *clientcmd.Config
-	TemplateFile       string
-	ReloadScript       string
+// routerConfig represents a container for all other router config types.
+// Any global config belongs on this level.
+type routerConfig struct {
+	// Config holds global flags that are passed to all client commands that need
+	// to connect to master.
+	Config *clientcmd.Config
+
+	// Type is the type of plugin being created: f5 or template.
+	Type string
+
+	// DefaultCertificate holds the certificate that will be used if no more
+	// specific certificate is found.  This is typically a wildcard certificate.
 	DefaultCertificate string
-	RouterService      ktypes.NamespacedName
-	StatsPort          string
-	StatsPassword      string
-	StatsUsername      string
+
+	// TemplateRouterConfig holds config items for template router plugins.
+	TemplateRouterConfig templateRouterConfig
+
+	// F5RouterConfig holds the configuration information for f5 router plugin.
+	F5RouterConfig f5RouterConfig
```

Comment:
I know its a bit too onerous to change how the config is used below + how that structure is passed around. But just a nit - it feels a bit odd to have a config structure with all the different implementing configs inside of it (f5 + template + anything else we add later). Think it would been better to have a specific config for the template router and one specifically for the f5 router and have them reuse (promoted fields) the same base struct containing the common fields (clientcmd config, type and default certificate).